### PR TITLE
Add required package for kerberos auth on jammy

### PR DESCRIPTION
**Pull request by chrisongthb**
This solves https://github.com/puppetlabs/puppetlabs-apache/issues/2402

### CODE REVIEW

**Review by ekohl — APPROVED** [review]
I've verified https://packages.ubuntu.com/jammy/libapache2-mod-auth-kerb exists, but packages.debian.org appears to be down now so I can't check it.

**Review by ekohl — COMMENTED** [review]
Looking at the diff, the you now define `nss` again, which previously wasn't. That package was dropped in Ubuntu 20.04.

I'm starting to wonder if it should be changed to this pattern:

```puppet
$_base_mod_packages = {
  # common ones
}

$_version_mod_packages = if VERSION_CHECK {
  {
    # ...
  }
} else {
  {}
}

$mod_packages = $_base_mod_packages + $_os_mod_packages
```

File: manifests/params.pp
Comment (ekohl) [inline manifests/params.pp:347]: Was already a problem previously, but this is only in Debian 9, not 10.

File: manifests/params.pp
Comment (ekohl) [inline manifests/params.pp:363]: Similar, this is only in 18.04.

**Review by ekohl — DISMISSED** [review]
This is indeed what I had in mind, and moving forward it'll hopefully be easier to maintain. I'll kick off CI so see if it agrees :)

**Comment by chrisongthb** [comment]
@ekohl if I got you right, it might work now.

**Comment by chrisongthb** [comment]
@ekohl could you check failed CI for SLES? I did not change anything regarding SLES.

**Comment by ekohl** [comment]
I think it's unrelated, but I don't know SLES. This needs approval from @puppetlabs/modules before it can be merged, so I'm now counting on them to take over.

**Comment by chrisongthb** [comment]
@LukasAud - rebase done :)

**Review by Ramesh7 — APPROVED** [review]

**Comment by Ramesh7** [comment]
Thanks @chrisongthb your contribution, The changes looks good. I am taking this forward.

**Pull request merged**
